src/kOS.Safe/Encapsulation/EnumerableValue.cs

```diff
@@ -1,26 +1,30 @@
-﻿using System.Collections;
+﻿using kOS.Safe.Encapsulation.Suffixes;
+using kOS.Safe.Serialization;
+using System.Collections;
 using System.Collections.Generic;
 using System.Linq;
-using kOS.Safe.Encapsulation.Suffixes;
-using kOS.Safe.Serialization;
 
 namespace kOS.Safe.Encapsulation
 {
     [kOS.Safe.Utilities.KOSNomenclature("Enumerable")]
-    public abstract class EnumerableValue<T, TC> : SerializableStructure, IEnumerable<T> where TC : IEnumerable<T> where T : Structure
+    public abstract class EnumerableValue<T, TE> : SerializableStructure, IEnumerable<T>
+        where TE : IEnumerable<T>
+        where T : Structure
     {
-        protected TC Collection { get; private set; }
+        protected TE InnerEnumerable { get; private set; }
+        private readonly string label;
 
-        protected EnumerableValue(TC collection)
+        protected EnumerableValue(string label, TE enumerable)
         {
-            Collection = collection;
+            this.label = label;
```

Comment:
hmm.  What might be better is to pass in the GetType() of the derived type rather than a string label.  Then set the string label using KOSNomenclature.GetKOSName().  Thus we ensure we are printing a message to the user using the type names that the rest of kOS is showing them for the type.